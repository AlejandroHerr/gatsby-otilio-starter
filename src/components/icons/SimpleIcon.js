import React from 'react';
import PropTypes from 'prop-types';

const SimpleIcon = ({ children, className, style }) => (
  <svg
    className={className}
    style={{
        fill: 'currentColor',
      height: '1.5rem',
      width: '1.5rem',
      ...style,
    }}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    {children}
  </svg>
);

SimpleIcon.defaultProps = {
  className: '',
  style: {},
};

SimpleIcon.propTypes = {
  children: PropTypes.func.isRequired,
  className: PropTypes.string,
  style: PropTypes.object,
};

export default SimpleIcon;
