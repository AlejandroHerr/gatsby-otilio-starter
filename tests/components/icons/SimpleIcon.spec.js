import React from 'react';
import { shallow } from 'enzyme';

import SimpleIcon from '../../../src/components/icons/SimpleIcon';

const defaultProps = {
  brand: 'testBrand',
};

const setup = (props = {}) => {
  const simpleIcon = shallow(<SimpleIcon {...defaultProps} {...props} />);

  return {
    simpleIcon,
  };
};

describe('icons', () => {
  describe('SimpleIcon', () => {
    it('should render a default SimpleIcon', () => {
      const { simpleIcon } = setup();

      expect(simpleIcon.hasClass(`simple_icon__${defaultProps.brand}`)).toBeTruthy();
      expect(simpleIcon).toMatchSnapshot();
    });
    it('should render a SimpleIcon with custom options', () => {
      const props = {
        className: 'testClassName',
        style: {
          color: 'black',
        },
      };
      const { simpleIcon } = setup(props);

      expect(simpleIcon.hasClass(`simple_icon__${defaultProps.brand}`)).toBeTruthy();
      expect(simpleIcon.hasClass(props.className)).toBeTruthy();
      expect(simpleIcon.find('svg').prop('style')).toBe(props.style);
      expect(simpleIcon).toMatchSnapshot();
    });
  });
});
