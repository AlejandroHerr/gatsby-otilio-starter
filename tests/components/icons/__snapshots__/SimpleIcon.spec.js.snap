// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons SimpleIcon should render a SimpleIcon with custom options 1`] = `
<svg
  className="simple_icon simple_icon__testBrand testClassName"
  style={
    Object {
      "color": "black",
    }
  }
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path />
</svg>
`;

exports[`icons SimpleIcon should render a default SimpleIcon 1`] = `
<svg
  className="simple_icon simple_icon__testBrand"
  style={Object {}}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path />
</svg>
`;
